add xfce-polkit

baronleonardo opened the pull request:
###### Description of changes

A simple PolicyKit authentication agent for XFCE

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Commit:
add xfce-polkit

Commit:
Update default.nix

add xfce-polkit to xfce

Commit:
Merge branch 'NixOS:master' into master

bobby285271 commented on pkgs/desktops/xfce/core/xfce-polkit/default.nix:
```diff
@@ -0,0 +1,40 @@
+
+  buildInputs = [
+    libxfce4ui
+    gtk3
```
`gtk3` is probably not a direct dependency of this package?

Commit:
use xfce-polkit as the default polkit for xfce

bobby285271 commented on pkgs/desktops/xfce/core/xfce-polkit/default.nix:
```diff
@@ -0,0 +1,40 @@
+    homepage = "https://github.com/ncopa/xfce-polkit";
+    license = [ licenses.gpl2Plus ];
+    platforms = platforms.unix;
+    maintainers = teams.xfce.members;
```
We will appreciate if you can also also list yourself in maintainers

baronleonardo replied:
done

bobby285271 reviewed (commented):
Did not test. Did not review the module change.

P.s.: We encourage writing commit messages according to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) since some users prefer avoiding squash merge :woman_shrugging:

baronleonardo commented on pkgs/desktops/xfce/core/xfce-polkit/default.nix:
```diff
@@ -0,0 +1,40 @@
+  };
+
+  nativeBuildInputs = [
+    glib
```
the project itself says that I need glib as a dependency
https://github.com/ncopa/xfce-polkit

bobby285271 replied:
Yeah we know it is a dependency. I think it should probably be moved to `buildInputs`, to have the same host (runtime) platform as the built package.

baronleonardo replied:
done

Commit:
Update maintainer-list.nix

Commit:
xfce-polkit: init at 0.3

- A simple PolicyKit authentication agent for XFCE
- maintainers: add baronleonardo

Commit:
maintainers: add baronleonardo

Commit:
xfce-polkit: init at 0.3

- A simple PolicyKit authentication agent for XFCE

Commit:
xfce-polkit: init at 0.3

- solve syntax issue with the maintainers
